### PR TITLE
Add Castra columnstore optimization

### PR DESCRIPTION
**Pull request by mrocklin**
This provides fast column access to castra objects.

This also adds castra as a dependency

cc @cpcloud

### CODE REVIEW

File: dask/dataframe/optimize.py [hunk 1]
```diff
@@ -19,7 +20,11 @@
         # Merge column access into bcolz loading
         RewriteRule((getitem, (dataframe_from_ctable, a, b, c, d), e),
                     (dataframe_from_ctable, a, b, e, d),
-                    (a, b, c, d, e)))
+                    (a, b, c, d, e)),
+        RewriteRule((getitem, (Castra.load_partition, '~c', '~part', '~cols1'),
```
Comment (jcrist) [inline dask/dataframe/optimize.py]: Adding this optimization, without ever inserting castra into dask ourselves makes this seem like an optimization that should only exist if castra is on the system (only add this rule if castra is importable?). Unless you plan to use castra in other places, and this is just the first one.

Comment (mrocklin) [inline dask/dataframe/optimize.py]: Yes, I plan to rip out `pframe` and replace it with a combination of `partd` and `castra`.

Sorry for the on-disk data structure churn.  I don't expect these to be the last iterations.

Comment (mrocklin) [inline dask/dataframe/optimize.py]: I've made this robust to castra's absence

**Comment by mrocklin** [comment]
Merging soon if no comment.

**Comment by jcrist** [comment]
Looks good to me.

**Pull request merged**